core/src/main/java/io/kestra/core/schedulers/AbstractScheduler.java

```diff
@@ -153,111 +181,173 @@ public void run() {
-                .map(trigger -> {
-                    RunContext runContext = runContextFactory.of(flow, trigger);
-
-                    return new FlowWithTrigger(
+                .map(abstractTrigger -> {
+                    RunContext runContext = runContextFactory.of(flow, abstractTrigger);
+                    ConditionContext conditionContext = conditionService.conditionContext(runContext, flow, null);
+                    Trigger triggerContext = null;
+                    try {
+                        Trigger lastTrigger = triggerContextsToEvaluate
+                            .stream()
+                            .filter(triggerContextToFind -> triggerContextToFind.uid().equals(Trigger.uid(flow, abstractTrigger)))
+                            .findFirst()
+                            .orElse(null);
+                        // If trigger is not found in triggers to evaluate, then we ignore it
+                        if (lastTrigger == null) {
+                            return null;
+                            // Backwards compatibility
+                        } else if (lastTrigger.getNextExecutionDate() == null) {
+                            triggerContext = lastTrigger.toBuilder()
+                                .nextExecutionDate(((PollingTriggerInterface) abstractTrigger).nextEvaluationDate(conditionContext, Optional.of(lastTrigger)))
+                                .build();
+                            this.triggerState.save(triggerContext, scheduleContext);
+                        } else {
+                            triggerContext = lastTrigger;
+                        }
+                    } catch (Exception e) {
+                        logError(conditionContext, flow, abstractTrigger, e);
+                        return null;
+                    }
+                    return new FlowWithTriggers(
                         flow,
-                        trigger,
+                        abstractTrigger,
+                        triggerContext,
                         runContext,
                         conditionService.conditionContext(runContext, flow, null)
                     );
                 })
             )
-            .toList();
+            .filter(Objects::nonNull).toList();
     }
 
+    abstract public void handleNext(ZonedDateTime now, BiConsumer<List<Trigger>, ScheduleContextInterface> consumer);
+
     private void handle() {
         if (!this.isReady) {
             log.warn("Scheduler is not ready, waiting");
         }
 
-        metricRegistry
-            .counter(MetricRegistry.SCHEDULER_LOOP_COUNT)
-            .increment();
+        // FIXME: New flow doesnt seems to pass in the flowListeners
+//        if (!this.isInitialized && !flowListeners.flows().isEmpty()) {
+        this.initializedTriggers(flowListeners.flows());
+//            this.isInitialized = true;
+//        }
```

Comment:
Well, I think this is because you remove the listen on the flow listener L104 ...

Reply:
right, remove the hack and restore the listener global